sandbox/libs/authn/docs/jwt.md

```diff
@@ -37,7 +36,96 @@ There are 2 types of tokens:
 - Refresh Token - A Refresh Token is longer lasting and used to receive short-lived Access Tokens that actually confer access
 - Access Token - Access Tokens are short-lived tokens that confer access
 
-Creation of an asynchronous job or task should require a Refresh Token on creation which is associated with a user and let's the job run with the user's permissions. When the task
-starts execution it will use the refresh token to obtain an access token which will allow the task to interact with the cluster as the user who the token was created for.
+Creation of an asynchronous job or task should require a Refresh Token on creation which is associated with a subject and let's the job run with the subject's permissions. When the task
+starts execution it will use the refresh token to obtain an access token which will allow the task to interact with the cluster as the subject who the token was created for.
+
+Refresh tokens will be associated with a subject (and an extension?) and both the subject and administrators (or subjects with requisite permissions to revoke tokens of others) will be able to revoke access to the tokens. Issuing a token for oneself or on behalf of others will be another set of permissions that can be assigned.
```

Comment:
I think this is where our principal on the Subject might need to transition from a singleton to collection, e.g.:

```java
List<Principal> = List.of(
   new StringPrincipal("InternalUser:Danny),
   new StringPrincipal("Extension:SudoRm")
);
```

So when doing subject.isAllowed("opensearch:admin/wreck/thisCluster"), we could verify that both the internal user Danny has the permissions AND that extension SudoRm has the permissions too.

Reply:
To accomplish extension specific tokens we will need to manage a secret for each extension to sign the token with.